pkg/apiserver/rest/usecase/addon.go

```diff
@@ -91,11 +91,11 @@ type addonUsecaseImpl struct {
 	apply      apply.Applicator
 }
 
-// GetAddon will get addon information, if detailed is not set, addon's componennt and internal definition won't be returned
+// GetAddon will get addon information, if detailed is not set, addon's component and internal definition won't be returned
 func (u *addonUsecaseImpl) GetAddon(ctx context.Context, name string, registry string, detailed bool) (*apis.DetailAddonResponse, error) {
 	addons, err := u.ListAddons(ctx, detailed, registry, "")
 	if err != nil {
-		return nil, err
+		log.Logger.Errorf("err when getting addon %s", name)
```

Comment:
If this error happens, it doesn't make any sense to continue the rest of the code.
Please return error